File: C#OOP-BasicsJune2016/1.DefiningClasses/CarSalesmanSolution2/Car.cs
using System;

public class Car
{
    public string model;
    public Engine engine;
    public int weight;
    public string color;

    public Car(string model, Engine engine, string color) : this(model, engine, -1, color)
    {

    }

    public Car(string model, Engine engine, int weight) : this(model, engine, weight, @"n/a")
    {

    }

    public Car(string model, Engine engine) : this(model, engine, -1, @"n/a")
    {

    }

    public Car(string model, Engine engine, int weight, string color)
    {
        this.model = model;
        this.engine = engine;
        this.weight = weight;
        this.color = color;
    }
}

File: NodeJSExpressJSJune2018/06.Sessions-Authentication-Excercise/server/models/Car.js
const mongoose = require('mongoose');

// mongoose validate messages can be customized with 
const REQUIRED_VALIDATION_MESSAGE = '{PATH} is required';

let carSchema = new mongoose.Schema({
    make: { type: String, required: true, REQUIRED_VALIDATION_MESSAGE },
    model: { type: String, required: true },
    price: { type: Number, required: true },
    imageUrl: { type: String, required: true },
    color: { type: String, default: 'random color' },
    isRented: { type: Boolean, default: false },
    timesRented: { type: Number, default: 0 }
});

let Car = mongoose.model('Car', carSchema);

module.exports = Car;